portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,25 +92,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         datastore.put(messageEntity); 
 
-        response.setContentType("text/html");
-
         //checking for empty messages
-        if(!clientMessage.isEmpty()){
+        List<String> messages = new ArrayList<String>();
+        if(!clientMessage.equals(" ")){
             messages.add(clientMessage+"\n");
         }
 
         //printing each individual message
         for(int i=0; i< messages.size(); i++) {
             response.getWriter().println(messages.get(i));
         }
-        //TODO:
-        //look at a better way to clean the boxes
-        //get the input # from html and send to JS without redirecting so that multiple users
-        //can use the site at once
-        response.sendRedirect("/index.html");
+        response.sendRedirect("/leaveamessage.html");
```

Comment:
Messages is only going to have 0 or 1 elements. Is that intended?

Reply:
No, I have fixed it to write to the client only if the message has content.